### DISCUSSION TITLE
Initial values for *_Intercept and *_1|participant_id are NaN or don't take effect in hierarchical model

### DISCUSSION THREAD

**Discussion by SaschaFroelich** [lnccbrown/HSSM#555] [Q&A]
Hi,

I am specifying a hierarchical model for 10 participants, like so for example (using `theta`here only for example purposes, the problem is the same for all variables):

```
hssm_model = hssm.HSSM(
    data = ddmdata,
    model = "angle",
    hierarchical = True,
    categorical = 'jokercondition',
    include=[
        {
              "name": "theta",
              "formula": "theta ~ 1 + (1|participant_id)",
              "prior": {
                  "Intercept": {
                      "name": "Uniform",
                      "lower": 0.3,
                      "upper": 0.7,
                      "initval": 0.8},
                  
                  "1|participant_id": {"name": "Normal", 
                                        "mu": {"name": "Normal", 
                                                "mu": 0, 
                                                "sigma": 0.5,
                                                "initval": 0.9},
                                        "sigma": {"name": "HalfNormal", "sigma": .3},
                                        "initval": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
              },
              "link": "identity",
          },
      ],
  )
```

However, as `hssm_model.initial_point()` reveals, while the initial value for `theta_1|participant_id_mu` did take effect, those for `theta_Intercept` and for the individual participants (I assume it's `theta_1|participant_id_offset`) did not (and the initial point for `theta_Intercept` is even `nan`:

```
{'v_Intercept': array(0.),
 'v_1|participant_id_sigma': array(0.27082359),
 'v_1|participant_id_offset': array([-0.00720059,  0.00269359, -0.0047719 ,  0.00153111,  0.00471117,
         0.00960931,  0.00806066, -0.00743628,  0.00386247, -0.0078001 ]),
 'a_Intercept': array(1.5),
 'a_1|participant_id_sigma': array(0.27082359),
 'a_1|participant_id_offset': array([ 0.00509311,  0.00909914, -0.00905317, -0.00765635, -0.00776619,
        -0.00816629,  0.0059612 , -0.00665181, -0.00723022,  0.00747891]),
 'z_Intercept': array(0.5),
 'z_1|participant_id_sigma': array(0.27082359),
 'z_1|participant_id_offset': array([-1.82387477e-03, -8.55270700e-05, -5.16602118e-03,  7.54557969e-03,
         6.87858090e-04,  5.24517521e-03,  6.61454396e-03,  7.68334139e-03,
         8.84313975e-03,  3.75443068e-03]),
 't_Intercept': array(0.025),
 't_1|participant_id_sigma': array(0.27082359),
 't_1|participant_id_offset': array([ 0.0041733 ,  0.00216332, -0.00055206,  0.00849784,  0.00319137,
         0.00156882,  0.00578121,  0.00156184, -0.00546336,  0.00181441]),
 'theta_Intercept': array(nan),
 'theta_1|participant_id_mu': array(0.9),
 'theta_1|participant_id_sigma': array(0.3),
 'theta_1|participant_id_offset': array([-0.00223896,  0.00641834, -0.00508269,  0.00181714, -0.00385215,
        -0.00608818,  0.0062197 ,  0.00155075,  0.00321892, -0.00957484])}
```

How can I set the individual initial values in such a way that they take effect?

**Comment by AlexanderFengler** [comment]
Hi @SaschaFroelich,

Few things:

1. The initval for `theta_Intercept` is in fact appropriately NaN in this case, since you specify the initial value outside of the legal bounds of the Uniform distribution for the parameter.
2. We made some improvements to the way users can specify and store initial values in HSSM
3. And we have added a new tutorial with focus on setting initial values in the main branch, to illustrate best/good practices on that front. (you can look at it already under docs/tutorials/initial_values.ipynb).

**Reply by SaschaFroelich** [reply]
Hi @AlexanderFengler,

thank you for your reply. Regarding 1: Oops, thank you, I completely missed that! However, the initital values for `1|participant_id` still don't take effect. In the new tutorial you say that all initial values except for `Intercept` should be set 0, is that why it automatically initializes the participant offsets to close to 0? Even so, I think there are cases where setting an initial value for the offsets might be warranted, like when the value for some participants converged after sampling, and I want to start sampling anew but this time with the previously converged parameters as initvals.

Best,
Sascha
